src/main/java/io/th0rgal/oraxen/items/OraxenMeta.java

```diff
@@ -40,6 +43,12 @@ public void setPackInfos(ConfigurationSection configurationSection) {
         this.pullingModels = configurationSection.isList("pulling_models")
                 ? configurationSection.getStringList("pulling_models") : null;
         this.layers = configurationSection.getStringList("textures");
+        ConfigurationSection texturesSection = configurationSection.getConfigurationSection("textures");
+        if (texturesSection != null) {
+            Map<String, String> layersMap = new HashMap<>();
+            texturesSection.getKeys(false).forEach(key -> layersMap.put(key, texturesSection.getString(key)));
+            this.layersMap = layersMap;
+        }

```

Comment:
realisticly it would be either or here. This would support both
Think it is better to make an if statement here, something like
```java
if (configurationSection.isStringList("textures")
    this.layers = configurationSection.getStringList("textures");
else this.layers = null;

// same for layersMap
```

Reply:
I think this `else` is unnecessary, bc `null` value of this field stays as it was, so what about something like this:
```java
if (configurationSection.isStringList("textures")
    this.layers = configurationSection.getStringList("textures");
ConfigurationSection texturesSection = configurationSection.getConfigurationSection("textures");
if (texturesSection != null) {
    Map<String, String> layersMap = new HashMap<>();
    texturesSection.getKeys(false).forEach(key -> layersMap.put(key, texturesSection.getString(key)));
    this.layersMap = layersMap;
}
```
or this:
```java
if (configurationSection.isStringList("textures")
    this.layers = configurationSection.getStringList("textures");
else if (configurationSection.getConfigurationSection("textures") != null) {
    ConfigurationSection texturesSection = configurationSection.getConfigurationSection("textures");
    Map<String, String> layersMap = new HashMap<>();
    texturesSection.getKeys(false).forEach(key -> layersMap.put(key, texturesSection.getString(key)));
    this.layersMap = layersMap;
}
```
And also, maybe `ModelGenerator` should have 
```java
} else if (oraxenMeta.hasLayers() {
```
at line 19?

Reply:
First example is the same, u either want layers or layerMap but never both
Second example looks better, but the else if should be isConfigurationSection
literally means nothing in the end but yeah
Otherwise looks good, though you could also .stream().map() the getKeys part instead of looping through and adding to a new list